### PR TITLE
Integrate submission page

### PR DESCRIPTION
**Pull request by junhonglow**
# Pull Request

## Description

/assignments/{id}/submission Integration
- Modify to now only reflect latest submission feedback from API

/assignments/submissions
- Student view has been modified to show all questions and whether a submission has been made for it.
- Teacher view has been modified to show all students and whether they have attempted the question.

## Screenshots (if applicable)
Student View /assignments/submissions
image

Tutor View /assignments/submissions
image

Student View AND Tutor View /assignments/{id}/submission
No errors
image
If there are errors
image

## Checklist

- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

### CODE REVIEW

File: frontend/src/app/assignments/submissions/page.tsx [hunk 1]
```diff
@@ -1,261 +1,97 @@
...
+      }
+      const submissionsData: Record<string, SubmissionData[]> = {};
+      if (user.role === "student") {
+        // Retrieve all assignments that are published
+        const assignments =  await assignmentService.getAssignmentsByUserId(
+          user.uid,
+          true,
+          true
+        );
+        for (const assignment of assignments) {
```
Comment (tryyang2001) [inline frontend/src/app/assignments/submissions/page.tsx:39]: Consider using `Promise.all` to send multiple API calls in parallel if they are not dependent of each other.

**Comment by tryyang2001** [comment]
@junhonglow kindly resolve #86 in this PR as well.

**Comment by tryyang2001** [comment]
image
You might want to remove the "Grades" tab as we are not grading, and also make the feedback more obvious, as for now I can only see the highlighted line but have no idea what goes wrong.

**Review by vivienherq — APPROVED** [review]

**Comment by junhonglow** [comment]
> @junhonglow kindly resolve #86 in this PR as well.

Just saw that vivien is currently handling that in a different PR so I will not do this in this PR.

File: backend/user-service/routes/user/user-route.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ router.delete("/deleteUser", auth, userController.deleteUser);
 router.delete("/clearCookie", userController.clearCookie);
 router.put("/updateUserPassword", auth, userController.updateUserPassword);
 router.put("/updateUserInfo", auth, userController.updateUserInfo);
+router.get("/getAllStudents", userController.getAllStudents);
```
Comment (hhchinh2002) [inline backend/user-service/routes/user/user-route.ts:12]: Imo, it would be a good idea to put `auth` here for `getAllStudents`

**Comment by junhonglow** [comment]
There are no longer an insane amount of API calls to backend, but rather depends on the number of assignments the user has.

**Comment by junhonglow** [comment]
image
Is this an ITS issue.

**Comment by tryyang2001** [comment]
> image

No cause Python doesn't recognize "false", "true", it should be "True" and "False". In some sense it can be the ITS issue as well haha.

**Pull request merged**